tests/vi_test.py

```diff
@@ -259,8 +259,9 @@ def test_pickle_support(q: str):
     )
     posterior.set_default_x(torch.tensor(np.zeros((num_dim,)).astype(np.float32)))
 
-    torch.save(posterior, "posterior.pt")
-    posterior_loaded = torch.load("posterior.pt")
+    filename = f"posterior_{q}.pt"
+    torch.save(posterior, filename)
```

Comment:
For such use case, it is usually better to use the  `tempfile` module with a context manager, which ensures that the file is clean up even if the test fails (which is currently not the case)

```python
from tempfile import Tempfile

with tempfile.NamedTemporaryFile(suffix=".pt") as f:
    torch.save(posterior, f.name)
    ...
# the file is deleted when existing the context even in case an error occurs
```

Reply:
That's a great suggestion, thanks.